pkg/cluster/workloadidentityresources_test.go

```diff
@@ -1240,6 +1241,16 @@ func TestEnsurePlatformWorkloadIdentityRBAC(t *testing.T) {
 						}
 						gotResources := template.Resources
 
+						sortResources := func(resources []*arm.Resource) {
+							sort.Slice(resources, func(i, j int) bool {
+								ri := resources[i].Resource.(mgmtauthorization.RoleAssignment)
+								rj := resources[j].Resource.(mgmtauthorization.RoleAssignment)
+								return *ri.Name < *rj.Name
+							})
+						}
+						sortResources(gotResources)
+						sortResources(tt.wantAdded)
+
 						if diff := cmp.Diff(gotResources, tt.wantAdded); diff != "" {
```

Comment:
I'm fine with the explicit sort, but the go-cmp package has a SortSlices option that could make this a little more terse: https://pkg.go.dev/github.com/google/go-cmp@v0.7.0/cmp/cmpopts#SortSlices